tests/foreman/ui/test_reporttemplates.py

```diff
@@ -342,3 +398,71 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
     :expectedresults: Report can't be downloaded. Error.
     :CaseImportance: High
     """
+
+
+@tier3
+def test_positive_gen_entitlements_reports_multiple_formats(session, content_setup, module_org):
+    """Generate reports using the Entitlements template in html, yaml, json, and csv format.
+
+        :id: b268663d-c213-4e59-8f81-61bec0838b1e
+
+
+        :setup: Installed Satellite with Organization, Activation key,
+                Content View, Content Host, and Subscriptions.
+
+        :steps:
+            1. Monitor -> Report Templates
+            2. Entitlements -> Generate
+            3. Click the dropdown to select output format
+            4. Submit
+
+        :expectedresults: reports are generated containing all the expected information
+                          regarding Entitlements for each output format.
+
+        :CaseImportance: High
+    """
+    with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+        vm.install_katello_ca()
+        module_org, ak = content_setup
+        vm.register_contenthost(module_org.label, ak.name)
+        assert vm.subscribed
+        with session:
+            session.location.select('Default Location')
+            result_json = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'JSON'}
+            )
+            with open(result_json) as json_file:
+                data_json = json.load(json_file)
+            assert any(entitlement['Name'] == vm.hostname for entitlement in data_json)
+            assert any(
+                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+                for entitlement in data_json
+            )
+            result_yaml = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'YAML'}
+            )
+            with open(result_yaml) as yaml_file:
+                data_yaml = yaml.load(yaml_file, Loader=yaml.FullLoader)
+            assert any(entitlement['Name'] == vm.hostname for entitlement in data_yaml)
+            assert any(
+                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+                for entitlement in data_yaml
+            )
+            result_csv = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'CSV'}
+            )
+            with open(result_csv) as csv_file:
+                data_csv = csv.DictReader(csv_file)
+                res = next(data_csv)
+            assert res['Name'] == vm.hostname
+            assert res['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
```

Comment:
Haven't noticed that before. Technically, it checks only first row of CSV. To be consistent with rest of assertions and check for these strings anywhere in file, you would need:
```python
    res = list(data_csv)
assert any(entitlement['Name'] == vm.hostname for entitlement in res)
assert any(
    entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
    for entitlement in res
)
```
(maybe with better name than `res`)